src/client/mod.rs

```diff
@@ -48,1243 +48,22 @@
 //! # fn main () {}
 //! ```
 
-use std::error::Error as StdError;
-use std::fmt;
-use std::mem;
-use std::time::Duration;
-
-use futures_channel::oneshot;
-use futures_util::future::{self, Either, FutureExt as _, TryFutureExt as _};
-use http::header::{HeaderValue, HOST};
-use http::uri::Scheme;
-use http::{Method, Request, Response, Uri, Version};
-
-use self::connect::{sealed::Connect, Alpn, Connected, Connection};
-use self::pool::{Key as PoolKey, Pool, Poolable, Pooled, Reservation};
-use crate::body::{Body, HttpBody};
-use crate::common::{exec::BoxSendFuture, lazy as hyper_lazy, task, Future, Lazy, Pin, Poll};
-use crate::rt::Executor;
-
 #[cfg(feature = "tcp")]
 pub use self::connect::HttpConnector;
 
-pub mod conn;
 pub mod connect;
-pub(crate) mod dispatch;
-mod pool;
-pub mod service;
 #[cfg(test)]
 #[cfg(feature = "runtime")]
 mod tests;
 
-/// A Client to make outgoing HTTP requests.
-pub struct Client<C, B = Body> {
-    config: Config,
-    conn_builder: conn::Builder,
-    connector: C,
-    pool: Pool<PoolClient<B>>,
-}
-
-#[derive(Clone, Copy, Debug)]
-struct Config {
-    retry_canceled_requests: bool,
-    set_host: bool,
-    ver: Ver,
-}
-
-/// A `Future` that will resolve to an HTTP Response.
-///
-/// This is returned by `Client::request` (and `Client::get`).
-#[must_use = "futures do nothing unless polled"]
-pub struct ResponseFuture {
```

Comment:
This type needs to still be accessible at `hyper::client::ResponseFuture`, to prevent a breaking change.

Reply:
Eep! Great catch. Fixed!

Reply:
Did you push the fix? I can't seem to see it...

Reply:
🤦🏽‍♂️

Reply:
Should be good now!